platform/default/src/mbgl/util/default_thread_pool.cpp

```diff
@@ -5,12 +5,14 @@
 
 namespace mbgl {
 
-ThreadPool::ThreadPool(std::size_t count) {
+ThreadPool::ThreadPool(std::size_t count, ThreadLifecycle _lifecycle) : lifecycle(std::move(_lifecycle)) {
     threads.reserve(count);
     for (std::size_t i = 0; i < count; ++i) {
         threads.emplace_back([this, i]() {
```

Comment:
Can you try passing copy to the thread?
`threads.emplace_back([this, i](ThreadLifecycle lifecycle) {...}, _lifecycle)`;
or 
`threads.emplace_back([this, i, _lifecycle]() mutable {...})`;